cms/djangoapps/contentstore/views/tests/test_organizations.py

```diff
@@ -32,3 +35,53 @@ def test_organization_list(self):
         self.assertEqual(response.status_code, 200)
         org_names = json.loads(response.content)
         self.assertEqual(org_names, self.org_short_names)
+
+
+@patch.dict('django.conf.settings.FEATURES', {'ORGANIZATIONS_APP': True})
+@override_switch(settings.ENABLE_EDLY_ORGANIZATIONS, active=True)
+class TestEdlyOrganizationListing(TestCase):
+    """Verify Organization listing behavior."""
+    @patch.dict('django.conf.settings.FEATURES', {'ORGANIZATIONS_APP': True})
+    def setUp(self):
+        super(TestEdlyOrganizationListing, self).setUp()
+        self.staff = UserFactory(is_staff=True)
+        self.client.login(username=self.staff.username, password='test')
+        self.org_names_listing_url = reverse('organizations')
+
+    def test_without_authentication(self):
+        """
+        Verify authentication is required when accessing the endpoint.
+        """
+        self.client.logout()
+        response = self.client.get(self.org_names_listing_url)
+        self.assertEqual(response.status_code, 302)
+
+    def test_organization_list(self):
+        """
+        Verify that the organization names list api returns list of organization short names.
+        """
+
+        studio_site = SiteFactory()
+        edly_organization = EdlyOrganizationFactory(name='Test Edly Organization Name')
+        edly_sub_organization = EdlySubOrganizationFactory(
+            name='Test Edly Sub Organization Name',
+            slug='test-edly-sub-organization-name',
+            studio_site=studio_site,
+            edly_organization=edly_organization
+        )
+
+        edx_organization = edly_sub_organization.edx_organization
+        edx_organization.short_name  = 'test-edx-organization'
+        edx_organization.save();
+
+        response = self.client.get(self.org_names_listing_url, HTTP_ACCEPT='application/json', SERVER_NAME=studio_site.domain)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json()[0], edly_sub_organization.edx_organization.short_name)
+
+
+        studio_site_2 = SiteFactory()
```

Comment:
Add a comment explaining why we are creating `studio_site_2`, e.g.
```
Now verify that if there is no "EdlySubOrganization" linked to a studio site the 
organization names list API returns empty response
```

Reply:
Still need to update comment as suggested

Reply:
The comments already added. Dont know why github not showing the difference.

Reply:
update comment